pkg/clients/clients.go

```diff
@@ -196,6 +198,33 @@ func (k *k8sClients) GetPlatform() (string, error) {
 	}
 }
 
+func (k *k8sClients) GetNodesByLabels(ctx context.Context, matchingLabels map[string]string) (*v1.NodeList, error) {
+	nodes, err := k.clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: labels.Set(matchingLabels).String()})
+	if err != nil {
+		return nil, err
+	}
+
+	// filter nodes by taints
+	nodesWithoutTaints := nodes.Items[:0]
+	for _, node := range nodes.Items {
+		if k.isNodeNotExecOrSchedule(&node) {
+			continue
+		}
+		nodesWithoutTaints = append(nodesWithoutTaints, node)
+	}
+	nodes.Items = nodesWithoutTaints
+	return nodes, nil
+}
+
+func (k *k8sClients) isNodeNotExecOrSchedule(node *v1.Node) bool {
+	for _, taint := range node.Spec.Taints {
+		if taint.Effect == "NoSchedule" || taint.Effect == "NoExecute" {
```

Comment:
Wondering if we can use the constants here:
```suggestion
		if taint.Effect == v1.TaintEffectNoSchedule || taint.Effect == v1.TaintEffectNoExecute {
```

Reply:
fixed